gbpservice/nfp/configurator/agents/generic_config.py

```diff
@@ -187,34 +235,97 @@ def handle_event(self, ev):
             # Invoke service driver methods based on event type received
             result = getattr(driver, "%s" % ev.id.lower())(context, kwargs)
         except Exception as err:
-            result = ("Failed to process %s request for %s service type. %s" %
-                      (ev.id, service_type, str(err).capitalize()))
-            LOG.error(result)
-        finally:
-            msg = {'receiver': const.ORCHESTRATOR,
-                   'resource': resource,
-                   'method': ev.id,
-                   'kwargs': [
-                              {
-                               'context': context,
-                               'resource': resource,
-                               'request_info': request_info,
-                               'result': result
-                               }
-                            ]
-                   }
-            if not notification_data:
-                notification_data.update(msg)
-            else:
-                data = {
-                        'context': context,
-                        'resource': resource,
-                        'request_info': request_info,
-                        'result': result
-                        }
-                notification_data['kwargs'].append(data)
+            import traceback
+            traceback.print_exc()
+            LOG.error("Failed to process ev.id=%s, ev=%s reason=%s" % (ev.id,
+                                                                       ev.data,
+                                                                       err))
+            result = FAILED
+
+        if ev.id == const.EVENT_CONFIGURE_HEALTHMONITOR:
+            if (kwargs.get('periodicity') == INITIAL and
```

Comment:
In handle_event() method we are giving max_times arg for initial type of health monitor. After max_times attempts if the ev object is still being polled then poll_event_cancel() will be invoked by framework and this is also indication that initial health check failed. And here we are sending health check failed notification